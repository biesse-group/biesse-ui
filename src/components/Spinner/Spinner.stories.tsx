import { type Meta, type StoryObj } from "@storybook/react";
import styled from "styled-components";

import { Spinner } from "./Spinner";

const SpinnerContainer = styled.div`
  display: flex;
  height: 100px;
  justify-content: center;
  align-items: center;
`;

const meta: Meta<typeof Spinner> = {
  title: "Loaders/Spinner",
  component: (props) => (
    <SpinnerContainer>
      <Spinner {...props} />
    </SpinnerContainer>
  ),
};

export default meta;

type Story = StoryObj<typeof Spinner>;

export const Default: Story = {};

export const CustomProps: Story = {};
CustomProps.args = {
  color: "green",
  thickness: "12px",
  duration: 1500,
};
