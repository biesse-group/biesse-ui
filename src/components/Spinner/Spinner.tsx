import React from "react";
import styled, { keyframes } from "styled-components";

const spinAnimation = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

export interface SpinnerProps {
  width?: string;
  height?: string;
  thickness?: string;
  color?: string;
  duration?: number;
}

const SpinnerBody = styled.div<SpinnerProps>`
  height: ${({ height }) => height ?? "4rem"};
  width: ${({ width }) => width ?? "4rem"};
  border: ${({ thickness }) => thickness ?? "4px"} solid #d1d5db;
  border-top-color: ${({ theme, color }) => color ?? theme.color.primary};
  border-radius: 50%;
  animation: ${spinAnimation} ${({ duration }) => (duration ? `${duration}ms` : "800ms")} linear
    infinite;
`;

export const Spinner = (props: SpinnerProps) => {
  return <SpinnerBody {...props} />;
};
