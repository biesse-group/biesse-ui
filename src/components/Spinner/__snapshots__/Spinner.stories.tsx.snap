// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loaders/Spinner CustomProps smoke-test 1`] = `
<div class="sc-gEvEer dIppSL">
  <div color="green"
       class="sc-aXZVg dINVZx"
  >
  </div>
</div>
`;

exports[`Loaders/Spinner Default smoke-test 1`] = `
<div class="sc-gEvEer dIppSL">
  <div class="sc-aXZVg KIkzN">
  </div>
</div>
`;
